react-instantsearch-native/getting-started/src/SearchBox.tsx

```diff
@@ -1,44 +1,54 @@
 import React, { useEffect, useRef, useState } from 'react';
 import { StyleSheet, View, TextInput } from 'react-native';
-import { useSearchBox, UseSearchBoxProps } from 'react-instantsearch';
+import { useSearchBox, UseSearchBoxProps } from 'react-instantsearch-core';
 
 type SearchBoxProps = UseSearchBoxProps & {
   onChange: (newValue: string) => void;
 };
 
 export function SearchBox({ onChange, ...props }: SearchBoxProps) {
   const { query, refine } = useSearchBox(props);
-  const [inputValue, setInputValue] = useState(query);
+  const [value, setValue] = useState(query);
   const inputRef = useRef<TextInput>(null);
 
-  function setQuery(newQuery: string) {
-    setInputValue(newQuery);
-    refine(newQuery);
-  }
+  // Track when the value coming from the React state changes to synchronize
+  // it with InstantSearch.
+  useEffect(() => {
```

Comment:
in the React InstantSearch codebase we managed to have a searchbox without effect, is that possible here too?

Reply:
Good eye, this works without them. I updated the search box in both examples.